natchez-datadog/src/main/scala/com/ovoenergy/effect/natchez/DatadogSpan.scala

```diff
@@ -19,20 +21,28 @@ import natchez.{Kernel, Span, TraceValue}
  */
 case class DatadogSpan[F[_]: Sync: Clock](
   names: SpanNames,
-  ids: SpanIdentifiers,
+  ids: Ref[F, SpanIdentifiers],
   start: Long,
   queue: Queue[F, SubmittableSpan],
   meta: Ref[F, Map[String, TraceValue]],
 ) extends Span[F] {
 
+  def updateTraceToken(fields: Map[String, TraceValue]): F[Unit] =
+    fields.get("traceToken").traverse {
```

Comment:
I think the name of the field should be configurable, maybe someone calls their trace tokens differently.

Reply:
fair point, but given that Tom's branch is crashing I guess this could be improved later?

Reply:
yes it definitely should - This isn't the only hardcoded use of `traceToken` though - a field named `traceToken` gets set as a tag on each span submitted.

I think this is probably due a rethink in the longer term to make things like this configurable but I'd need more time. This whole module could even be replaced with the Datadog integration now present in the main Natchez repo but I don't know whether it supports absolutely everything this one does.

Reply:
Yup, that's fair too, thanks for clarifying it!